rust/src/transformations/make_stable_expr/expr_cast/mod.rs

```diff
@@ -0,0 +1,78 @@
+use polars::chunked_array::cast::CastOptions;
+use polars::prelude::*;
+use polars_plan::dsl::Expr;
+
+use crate::core::{Function, MetricSpace, StabilityMap, Transformation};
+use crate::domains::{ExprDomain, OuterMetric, SeriesDomain};
+use crate::error::*;
+use crate::polars::ExprFunction;
+use crate::transformations::DatasetMetric;
+
+use super::StableExpr;
+
+#[cfg(test)]
+mod test;
+
+/// Make a Transformation that returns a `cast(dtype)` expression for a LazyFrame.
+///
+/// # Arguments
+/// * `input_domain` - Expr domain
+/// * `input_metric` - The metric under which neighboring LazyFrames are compared
+/// * `expr` - The clipping expression
+pub fn make_expr_cast<M: OuterMetric>(
+    input_domain: ExprDomain,
+    input_metric: M,
+    expr: Expr,
+) -> Fallible<Transformation<ExprDomain, ExprDomain, M, M>>
+where
+    M::InnerMetric: DatasetMetric,
+    M::Distance: Clone,
+    (ExprDomain, M): MetricSpace,
+    Expr: StableExpr<M, M>,
+{
+    let Expr::Cast {
+        expr: input,
+        data_type: to_type,
+        mut options,
+    } = expr
+    else {
+        return fallible!(MakeTransformation, "expected function expression");
+    };
+
+    // strict casting makes the transformation unstable: errors tell you things about private data
+    // Could throw an error if strict, but it is the default,
+    // so for ease-of-use it is forced to be non-strict
+    if let CastOptions::Strict = options {
+        options = CastOptions::NonStrict;
+    }
+
+    let t_prior = input
+        .as_ref()
+        .clone()
+        .make_stable(input_domain.clone(), input_metric.clone())?;
+    let (middle_domain, middle_metric) = t_prior.output_space();
+
+    let mut output_domain = middle_domain.clone();
+    let series_domain = output_domain.active_series_mut()?;
+    let name = series_domain.field.name.as_ref();
+
+    // it is possible to tighten this:
+    // in cases where casting will never fail, the nullable and/or nan bits can be left false
+    // in the meantime, users will need to impute
```

Comment:
Do we want an issue to track this? Perhaps not, if it's low priority and really just a reminder of work that could be done if we're in the neighborhood.